tests/kernel/wave/attention/paged_attention_test.py

```diff
@@ -172,88 +97,134 @@ def ref_paged_attn(
+    kv_lens = torch.load(os.path.join(directory, "kv_lens.pt"))
+    return query, key_cache, value_cache, block_table, request_indices, kv_lens
+
+
 # TODO: Investigate errors on MI250.
 @require_e2e
 @require_cdna3
-@pytest.mark.parametrize("seq_lens", SEQ_LENS)
-@pytest.mark.parametrize("num_heads", NUM_HEADS)
-@pytest.mark.parametrize("head_size", HEAD_SIZES)
-@pytest.mark.parametrize("block_size", BLOCK_SIZES)
-@pytest.mark.parametrize("sliding_window", [None])
-@pytest.mark.parametrize("dtype", DTYPES)
-@pytest.mark.parametrize("soft_cap", [None])
-@pytest.mark.parametrize("num_blocks", NUM_BLOCKS)
+@pytest.mark.parametrize("shape", shapes)
+@pytest.mark.parametrize("dtype", [torch.float16])
 @pytest.mark.parametrize("enable_scheduling", [False])
+@pytest.mark.parametrize("num_kv_splits", [8])
 @pytest.mark.parametrize(
     "mfma_variant",
     [
         MMAType.F32_16x16x16_F16,
     ],
 )
 def testPagedFlashDecoding(
-    seq_lens: List[tuple[int, int]],
-    num_heads: tuple[int, int],
-    head_size: int,
-    block_size: int,
-    sliding_window: Optional[int],
+    shape: tuple[int],
     dtype: torch.dtype,
-    soft_cap: Optional[float],
-    num_blocks: int,
     enable_scheduling: bool,
+    num_kv_splits: int,
     mfma_variant: MMAType,
     request,
 ):
 
     torch.manual_seed(0)
-    num_seqs = len(seq_lens)
-    query_lens = [x[0] for x in seq_lens]
-    kv_lens = [x[1] for x in seq_lens]
-    num_query_heads = num_heads[0]
-    num_kv_heads = num_heads[1]
-    assert num_query_heads % num_kv_heads == 0
-    max_query_len = max(query_lens)
-    max_kv_len = max(kv_lens)
-    window_size = (sliding_window - 1, 0) if sliding_window is not None else (-1, -1)
-    scale = head_size**-0.5
-
-    query = device_randn(sum(query_lens), num_query_heads, head_size, dtype=dtype)
-    key_cache = device_randn(
-        num_blocks, block_size, num_kv_heads, head_size, dtype=dtype
-    )
-    value_cache = torch.randn_like(key_cache)
-    # TODO: The block table entries should be able to be a random number
-    # in the range [0, num_blocks * block_size), but that fails for now.
-    # As a workaround, the maximum value is set to num_seqs - 1.
-    block_table = device_randint(
-        0, num_blocks, (num_seqs, num_blocks), dtype=torch.int32
+    shape = paged_decode_attention_shape(
+        num_query_heads=shape[0],
+        num_kv_heads=shape[1],
+        head_size=shape[2],
+        head_size_kv=shape[3],
+        block_size=shape[4],
+        num_seqs=shape[5],
+        kv_lens=shape[6],
     )
-    request_indices = device_arange(num_seqs, dtype=torch.int32)
-    kv_lens_tensor = device_zeros(num_seqs, dtype=torch.int32)
-    for i in range(len(kv_lens)):
-        kv_lens_tensor[i] = kv_lens[i]
+    assert shape.num_query_heads % shape.num_kv_heads == 0
+    scale = shape.head_size**-0.5
+
+    artifact_directory = None
+    if not artifact_directory:
```

Comment:
? How could this not be `None` if you just set it to `None` here?

Reply:
This was just for debugging purposes to compare against inputs from sglang.